src/consensus/tx_verify.cpp

```diff
@@ -325,6 +348,32 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state, int nHeig
     return true;
 }
 
+inline bool checkOutput(const CTxOut& out, CValidationState& state, CAmount& nValueIn, std::map<std::string, CAmount>& nAssetVin, bool isV17active)
+{
+    // Check for negative or overflow values
+    nValueIn += out.nValue;
+    if (!MoneyRange(out.nValue, isV17active) || !MoneyRange(nValueIn, isV17active)) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-txns-values-outofrange");
+    }
+
+    if(out.scriptPubKey.IsAssetScript()){
+        CAssetTransfer assetTransfer;
```

Comment:
need to check somewhere in here that if this is asset script of any kind, out.nValue should always be 0

Reply:
fixed